### PR TITLE
Remove boost for case conversion and trim

### PR DESCRIPTION
**Pull request by twollgam**
### What does this PR do?

The pr removes the dependency for boosts case conversion. It is part of issue #39.

### Closes Issue(s)

None, but is part of #39.

### Motivation

Remove dependency of boost.

### CODE REVIEW

**Review by Baklap4 — APPROVED** [review]
LGTM, @ooshlablu any chance you can test this PR and see if we don't run into weird errors regarding case conversion. Some things to check out is the cache directory, and if all different songparsers still work correctly

**Comment by ooshlablu** [comment]
I tested this with guitar/bass and vocals using the AppImage and debs on Ubuntu 20.04 & 22.04, and everything looks fine to me. Game play seems to be unchanged, search seems to work how it did before, setting config options seems fine, and all of my songs are found and the ones I tried work fine.

LGTM

**Comment by twollgam** [comment]
@Baklap4 please review the issue again. I think I rebased the pr what causes the need for a new review :-(

File: game/util.cc [hunk 1]
```diff
@@ -32,3 +36,112 @@ std::string format(std::chrono::seconds const& unixtime, std::string const& form
...
+	return s;
+}
+
+std::string replace(std::string const& s, char from, char to) {
+	auto result = s;
+
+	return replace(result, from, to);
+}
+
+std::string& replace(std::string& s, char from, char to) {
```
Comment (Baklap4) [inline game/util.cc:80]: Might be better to add a formatter whether to ignore case or not. Now this won't replace things: `replace('a', 'A')` and it's not clear why.

File: game/util.cc [hunk 2]
```diff
@@ -32,3 +36,112 @@ std::string format(std::chrono::seconds const& unixtime, std::string const& form
...
+	return result;
+}
+
+std::string erase(std::string const& s, std::string const& toErase) {
+	auto result = s;
+
+	return erase(result, toErase);
+}
+
+std::string erase(std::string& s, std::string const& toErase) {
```
Comment (Baklap4) [inline game/util.cc:65]: Might be better to add  a formatter (see below)

File: game/util.cc [hunk 3]
```diff
@@ -32,3 +36,112 @@ std::string format(std::chrono::seconds const& unixtime, std::string const& form
...
+std::string toLower(std::string const& s) {
+	//return boost::algorithm::to_lower_copy(s);
+	auto result = s;
+
+	std::for_each(result.begin(), result.end(), [](auto& c){ c = static_cast<char>(std::tolower(c));});
+
+	return result;
+}
+
+std::string toUpper(std::string const& s) {
```
Comment (Baklap4) [inline game/util.cc:50]: Might be better to add  a locale since toUpper could do something else with weird characters.

**Review by Baklap4 — CHANGES_REQUESTED** [review]
Some minor changes to the util functions. We might have to consider ignore case and locales and right to left languages

File: game/util.cc [hunk 4]
```diff
@@ -32,3 +36,112 @@ std::string format(std::chrono::seconds const& unixtime, std::string const& form
...
+	return s;
+}
+
+std::string trimLeft(std::string const& s, std::locale const& locale) {
+	auto const start = std::find_if(s.begin(), s.end(), [&locale](char c){return !std::isspace(c, locale);}) - s.begin();
+
+	return s.substr(start);
+}
+
+std::string& trimLeft(std::string& s, std::locale const& locale) {
```
[the remark this answers is not in the document]
Comment (twollgam) [reply game/util.cc:123]: Good idea 👍 But I prefer front and back.

File: game/util.cc (hunk 1 above)
Comment (twollgam) [reply game/util.cc]: Don't know if I changed it as you expected. Please check that.

**Review by Baklap4 — APPROVED** [review]
Looking good to get rid of Boost!

Let's get this tested, but codewise looking good!

**Comment by BWagener** [comment]
Tested on windows, looks good to me.

**Comment by Baklap4** [comment]
Pipeline fails and conflicts

**Review by Baklap4 — APPROVED** [review]

**Pull request merged**